Updates/0216_q1268_afterscript.sql

```diff
@@ -0,0 +1,59 @@
+-- ---------------------------------------------------------
+-- CompleteScript for quest Suspicious Hoofprints q.1268
+-- ---------------------------------------------------------
+
+-- Quest template
+UPDATE quest_template SET CompleteScript=1268 WHERE entry=1268;
+
+-- Add scripts
+DELETE FROM `dbscripts_on_quest_end` WHERE `id`=1268;
+INSERT INTO `dbscripts_on_quest_end` (`id`, `delay`, `priority`, `command`, `datalong`, `datalong2`, `datalong3`, `buddy_entry`, `search_radius`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `comments`) VALUES 
+(1268, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2000020135, 0, 0, 0, 0, 0, 0, 0, 'Krog say'),
+(1268, 50, 0, 10, 4972, 0, 0, 0, 0, 0, 1, 0, 0, 0, -3155.99, -2898.27, 33.9313, 0.501176, 'Spawn Kagaro'),
+(1268, 300, 0, 20, 2, 1, 0, 4972, 20, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Start moving Kagaro'),
+(1268, 3500, 0, 0, 0, 0, 0, 0, 0, 0, 2000020136, 0, 0, 0, 0, 0, 0, 0, 'Krog say 2'),
+(1268, 4900, 0, 0, 0, 0, 0, 4972, 20, 7, 2000020137, 0, 0, 0, 0, 0, 0, 0, 'Kagoro say'),
+(1268, 5300, 0, 2, 66, 0, 0, 4972, 20, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Kagoro Salute'),
+(1268, 6000, 0, 20, 2, 2, 0, 4972, 20, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Kagoro run away');
+
+-- Add movement paths
+DELETE FROM `creature_movement_template` WHERE `entry`=4972;
+INSERT INTO `creature_movement_template` (`entry`, `pathId`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `waittime`, `script_id`, `comment`) VALUES 
+(4972, 1, 0, -3150.74, -2895.39, 34.1223, 0, 0, 0, NULL),
+(4972, 1, 1, -3145.96, -2890.87, 34.2473, 0, 0, 0, NULL),
+(4972, 1, 2, -3147.34, -2888.67, 34.2473, 2.5199, 0, 0, NULL),
+(4972, 1, 3, -3147.34, -2888.67, 34.2473, 2.5199, 300, 11625211, NULL),
+(4972, 2, 0, -3142.56, -2882.04, 34.7061, 0, 0, 0, NULL),
+(4972, 2, 1, -3140.06, -2878.79, 34.9561, 0, 0, 0, NULL),
+(4972, 2, 2, -3140.74, -2879.63, 34.9187, 0, 0, 0, NULL),
+(4972, 2, 3, -3145.74, -2863.88, 34.4187, 0, 0, 0, NULL),
+(4972, 2, 4, -3146.1, -2863.93, 34.5875, 0, 0, 0, NULL),
+(4972, 2, 5, -3152.85, -2861.43, 34.5875, 0, 0, 0, NULL),
+(4972, 2, 6, -3156.09, -2860.36, 34.7205, 0, 0, 0, NULL),
+(4972, 2, 7, -3163.84, -2859.11, 34.2205, 0, 0, 0, NULL),
+(4972, 2, 8, -3164.22, -2858.86, 34.5724, 0, 0, 0, NULL),
+(4972, 2, 9, -3179.65, -2855.16, 36.2173, 0, 0, 0, NULL),
+(4972, 2, 10, -3182.4, -2854.91, 37.7173, 0, 0, 0, NULL),
+(4972, 2, 11, -3183.9, -2854.41, 38.2173, 0, 0, 0, NULL),
+(4972, 2, 12, -3186.9, -2853.66, 40.2173, 0, 0, 0, NULL),
+(4972, 2, 13, -3188.9, -2853.16, 41.7173, 0, 0, 0, NULL),
+(4972, 2, 14, -3190.31, -2852.53, 42.4347, 0, 0, 0, NULL),
+(4972, 2, 15, -3192.81, -2851.28, 43.4347, 0, 0, 0, NULL),
+(4972, 2, 16, -3199.31, -2848.78, 41.9347, 0, 0, 0, NULL),
+(4972, 2, 17, -3202.06, -2847.78, 38.9347, 0, 0, 0, NULL),
+(4972, 2, 18, -3202.06, -2847.78, 38.9347, 0, 300, 11625210, NULL);
```

Comment:
A few things to note here: 
* When you add waypoints you should only copy the `Points` from sniff, not `Waypoints`.  `[0] Points: X: -3140.792 Y: -2866.917 Z: 34.66483` for example
Replace your waypoints with this:
```sql
-- Add movement paths
UPDATE creature_template SET MovementType=2 WHERE entry=4972;
DELETE FROM `creature_movement_template` WHERE `entry`=4972;
INSERT INTO `creature_movement_template` (`entry`, `pathId`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `waittime`, `script_id`, `comment`) VALUES 
(4972, 0, 1, -3147.338, -2888.669, 34.24735, 2.423081, 1000, 3, NULL),

(4972, 1, 1, -3140.792, -2866.917, 34.66483, 100, 0, 0, NULL),
(4972, 1, 2, -3149.111, -2862.073, 34.17603, 100, 0, 0, NULL),
(4972, 1, 3, -3162.162, -2859.5, 34.42603, 100, 0, 0, NULL),
(4972, 1, 4, -3171.867, -2857.081, 34.07432, 100, 0, 0, NULL),
(4972, 1, 5, -3184.27, -2854.473, 38.34727, 100, 0, 0, NULL),
(4972, 1, 6, -3190.406, -2852.848, 42.3148, 100, 0, 0, NULL),
(4972, 1, 7, -3202.731, -2847.48, 38.70749, 100, 1000, 1, NULL);
```
* If you look in the CreateObject2 packet you will see a row with the creature's destination `[4] Destination: X: -3147.338 Y: -2888.669 Z: 34.24735`. This is always the first point and it means the creature moves on waypoints on spawn. Therefore this row should be removed `(1268, 300, 0, 20, 2, 1, 0, 4972, 20, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Start moving Kagaro'),` and instead you should set MovementType=2 in the creature's creature_template to make the creature immediately start moving on the path. `UPDATE creature_template SET MovementType=2 WHERE entry=4972;`
Additionally, in the paste above I changed the pathIds to 0 and 1. 0 is the default path that will always be run on spawn if MovementType=2.
This can be manipulated by setting datalong3 to another value as seen below, but that is not needed in this script.

`SCRIPT_COMMAND_TEMP_SPAWN_CREATURE : "datalong3: The pathId you want the creature to immediately start moving on (if MovementType=2)"`

* `(1268, 5300, 0, 2, 66, 0, 0, 4972, 20, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Kagoro Salute'),` - This should be command 1 instead of 2.
* It is good practice to always set a despawntime in SCRIPT_COMMAND_TEMP_SPAWN_CREATURE  datalong2. This ensures that if something malfunctions there will not be a huge amount of NPCs spawned without a despawntimer, slowing down performance of the core. 
* The delay timers are not quite correct. The event finishes almost twice as fast as the timestamps in the sniff indicate. I have changed the timers and fixed the other issues from above:
```sql
-- Add scripts
DELETE FROM `dbscripts_on_quest_end` WHERE `id`=1268;
INSERT INTO `dbscripts_on_quest_end` (`id`, `delay`, `priority`, `command`, `datalong`, `datalong2`, `datalong3`, `buddy_entry`, `search_radius`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `comments`) VALUES 
(1268, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2000020135, 0, 0, 0, 0, 0, 0, 0, 'Krog say'),
(1268, 0, 0, 10, 4972, 100000, 0, 0, 0, 0, 1, 0, 0, 0, -3155.991, -2898.267, 33.93134, 0.501176, 'Spawn Kagaro'),
(1268, 5500, 0, 0, 0, 0, 0, 0, 0, 0, 2000020136, 0, 0, 0, 0, 0, 0, 0, 'Krog say 2'),
(1268, 7000, 0, 0, 0, 0, 0, 4972, 20, 7, 2000020137, 0, 0, 0, 0, 0, 0, 0, 'Kagoro say'),
(1268, 9000, 0, 1, 66, 0, 0, 4972, 20, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Kagoro Salute'),
(1268, 13000, 0, 20, 2, 1, 0, 4972, 20, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Kagoro run away');
```